allennlp/data/dataset.py

```diff
@@ -152,11 +151,11 @@ def as_arrays(self,  # ignore type
                 # This is creating a dict of {namespace: batch_array} for each
                 # namespace within this field. This is mostly utilised by TextFields,
                 # which will have one namespace per TokenIndexer used within the Field.
-                namespace_batch_dict = defaultdict(list)
+                namespace_batch_dict = defaultdict(list)  # type: Dict[str, List[numpy.array]]
                 for namespace_dict in field_array_list:
                     for namespace, array in namespace_dict.items():
                         namespace_batch_dict[namespace].append(array)
-                field_arrays[field_name] = {namespace: numpy.asarray(array_list) for
+                field_arrays[field_name] = {namespace: numpy.asarray(array_list) for  # type: ignore
```

Comment:
@joelgrus - I ignored the type here, because we are modifying this dictionary in place and it starts off with a different type for its values. Maybe I could change the type above to be a union of the possible types, is that better?